### PR TITLE
LOCO Upgrade

### PR DESCRIPTION
**Pull request by murilobalves**
- LOCOConfig: Now the algorithm setup is selected with the help of class constants in namedtuple format:
    - INVERSION: Normal or Transpose - Select if the jacobian to be inverted will be J or J.T * J
    - MINIMIZATION: Gauss-Newton or Levenberg-Marquardt - Select the minimization algorithm to be used, note: Levenberg-Marquardt works only using transpose inversion
    - SVD: Selection or Threshold - Select if the singular values selection in the jacobian will be by selecting the number of SV to keep or removing the low SV which are lower than SV_max * Threshold

- LOCO: The SV selection approach is changed: the old code selected the SV of 1/S matrix, where S is singular matrix found by J = U * S * V.T in the normal inversion case and in the transpose inversion case S is obtained by J.T * J = U * S * V.T. 
Now this code selects the SV of the S itself, then reconstructs the jacobian with this filtered singular matrix. After that, no other SV selection must be done. This is useful to keep the normal and transpose inversion methods equivalent with respect of singular values selection, which seems to be the proper approach.

### CODE REVIEW

File: apsuite/loco/main.py [hunk 1]
```diff
@@ -16,8 +16,8 @@ class LOCO:
     DEFAULT_TOL = 1e-3
```
Comment (xresende) [inline apsuite/loco/main.py:16]: these constants have units? if they so, I think we should include a comment in the line with it. I always use the format
```python

FREQ_DELTA = 10.0  # [Hz]

File: apsuite/loco/config.py [hunk 2]
```diff
@@ -21,8 +19,18 @@ class LOCOConfig:
...
     FAMNAME_RF = 'SRFCav'
 
+    INVERSION = _namedtuple(
+        'Methods', ['Normal', 'Transpose'])(0, 1)
+    MINIMIZATION = _namedtuple(
+        'Methods', ['GaussNewton', 'LevenbergMarquardt'])(0, 1)
+    SVD = _namedtuple(
+        'Methods', ['Selection', 'Threshold'])(0, 1)
+
     def __init__(self, **kwargs):
```
Comment (xresende) [inline apsuite/loco/config.py:29]: nice. I think the interface to LOCO config is becoming very clean.

File: apsuite/loco/main.py (hunk 1 above)
Comment (xresende) [reply apsuite/loco/main.py:16]: @murilobalves , I don't know why but when I make a comment on a code line, the line that shows ups in this conversation is not necessarily the correct one. I must be misusing it.

I was looking at `DEFAULT_DELTAK_NORMALIZATION`... but please check all constants.

Comment (murilobalves) [reply apsuite/loco/main.py:16]: Actually `DEFAULT_DELTAK_NORMALIZATION` is a parameter I would like to discuss with the group. I guess its unit will also be a result from this discussion. I just set it to 1.0 temporarily. The other class constants in LOCO are dimensionless. The constants in LOCOConfig has a comment indicating its units as you suggested.

Comment (xresende) [reply apsuite/loco/main.py:16]: all right, let's discuss in gitter... aproving this PR.

**Pull request merged**